### PR TITLE
New API for IBM Connections Cloud, extends scribejava-apis

### PR DESCRIPTION
**Pull request by DanieleVistalli**
I learned about scribejava and immediately loved it. So clear, so smart I couldn't avoid moving my apps to it.

As I work with IBM Connections Cloud services I created a new API object that covers integration needs for Connections Cloud.

It would be loverly if this can be integrated in the official repository.

If you've any request just let me know.

### CODE REVIEW

**Comment by kullfar** [comment]
Hi, nice to hear that.
1.every XXXApi.java have corresponding XXXExample.java. It's needed for ConnectionsCloudApi.
2.I will need test application credentials and test user credentials to run the Example
3.There're a lot of checktyle violations. Primarily for using tab character instead of spaces

File: scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+/**
+ * ConnectionsCloudApi supports OAuth2 endpoint for IBM Connections Cloud.
+ * 
+ * as IBM CC has multiple datacenter and multiple endpoints an enum object is provided 
+ * with the listing of allowable datacenters.
+ * 
+ * This API is based on BitBuckets Oauth2 API and simplifies usage of ConnectionsCloud
+ * services.
+ *  
+ * @author daniele.vistalli@factor-y.com
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java]: git blame is enough

File: scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+import java.security.InvalidParameterException;
+
+import com.github.scribejava.core.builder.api.DefaultApi20;
+import com.github.scribejava.core.extractors.OAuth2AccessTokenExtractor;
+import com.github.scribejava.core.extractors.TokenExtractor;
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import com.github.scribejava.core.model.Verb;
+
+/**
+ * ConnectionsCloudApi supports OAuth2 endpoint for IBM Connections Cloud.
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java]: Is it better to name this class like
IBMConnectionsCloudApi
?

File: scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			return InstanceHolderAP.INSTANCE;
+		default:
+			throw new InvalidParameterException("Datacenter name is not allowed: " + datacenter);
+		}
+    }
+
+    @Override
+    public Verb getAccessTokenVerb() {
+        return Verb.POST;
+    }
```
Comment (kullfar) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java]: no need to override this method. DefaultApi returns Verb.POST by default

Comment (DanieleVistalli) [inline scribejava-apis/src/main/java/com/github/scribejava/apis/ConnectionsCloudApi.java]: I do Fixes to clean up  and rename the class to IBMConnectionsCloudApi, then I'll git blame.

Thanks for the prompt response.

**Comment by kullfar** [comment]
outdated

**Pull request closed without merging**